mountpoint-s3-client/tests/put_object.rs

```diff
@@ -450,6 +451,7 @@ async fn check_sse(
 #[test_case(Some("aws:kms:dsse"), Some(get_test_kms_key_id()))]
 #[test_case(Some("aws:kms:dsse"), None)]
 #[test_case(None, None)]
+#[test_case(Some("AES256"), None)]
```

Comment:
Can you specify `kms_key_id` here? If not, how does it fail?

Reply:
no, it's not allowed by s3 api, it fails on `PutObjectRequest:write`:

```
ClientError(ResponseError(MetaRequestResult { response_status: 400, crt_error: Error(14343, "aws-c-s3: AWS_ERROR_S3_INVALID_RESPONSE_STATUS, Invalid response status from request"), error_response_headers: Some(Headers { inner: 0x7f10c00175d0 }), error_response_body: Some("<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n<Error><Code>InvalidArgument</Code><Message>Server Side Encryption with AWS KMS managed key requires HTTP header x-amz-server-side-encryption : aws:kms</Message><ArgumentName>x-amz-server-side-encryption</ArgumentName><ArgumentValue>AES256</ArgumentValue>...</Error>") }))
```

does it make sense to put a client-side check for this? (e.g. [here](https://github.com/awslabs/mountpoint-s3/blob/main/mountpoint-s3-client/src/s3_crt_client/put_object.rs#L57))